small improvements for tut1.rst (#17935)
nim-lang/Nim: doc/tut1.rst

  let input = readLine(stdin)   # works

The assignment statement
========================

The assignment statement assigns a new value to a variable or more generally
to a storage location:

.. code-block::
  var x = "abc" # introduces a new variable `x` and assigns a value to it
  x = "xyz"     # assigns a new value to `x`

`=` is the *assignment operator*. The assignment operator can be
overloaded. You can declare multiple variables with a single assignment
statement and all the variables will have the same value:

.. code-block::
    :test: "nim c $1"
  var x, y = 3  # assigns 3 to the variables `x` and `y`
  echo "x ", x  # outputs "x 3"
  echo "y ", y  # outputs "y 3"
  x = 42        # changes `x` to 42 without changing `y`
  echo "x ", x  # outputs "x 42"
  echo "y ", y  # outputs "y 3"

Note that declaring multiple variables with a single assignment that calls a
procedure can have unexpected results: the compiler will *unroll* the
assignments and end up calling the procedure several times. If the result of
the procedure depends on side effects, your variables may end up having
different values! For safety use side-effect-free procedures if making multiple

> assignments.

Reviewer: Remove this paragraph, it doesn't belong to the basic tutorial.

        return
      result = result + i

>   echo sumTillNegative() # echoes 0

Reviewer 2: ```suggestion
  assert sumTillNegative() == 0
```
simpler + more idiomatic, ditto below

Reviewer: But then the reader must already know about `assert`.

Reviewer 2: assert is pervasive in programming languages though, it's reasonable to expect users would get its meaning (but the tutorial can add a comment on the 1st instance of `assert`)

Author: I would agree with this change if this were some test in the stdlib.
This being a tutorial, I prefer `echo` in all the examples where is currently.

(Note that `toString` is usually the `$ <dollars.html>`_ operator in
Nim.) The compiler chooses the most appropriate proc for the `toString`
calls. How this overloading resolution algorithm works exactly is not

> discussed here. However, it does not lead to nasty surprises and is based
> on a quite simple unification algorithm. Ambiguous calls are reported as errors.

Reviewer: ```suggestion
discussed here -- see the manual for details. Ambiguous calls are reported as errors.
```
